web-bundle/src/main/java/com/graphhopper/swl/CustomCarFlagEncoder.java

```diff
@@ -0,0 +1,42 @@
+package com.graphhopper.swl;
+
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.routing.util.CarFlagEncoder;
+import com.graphhopper.util.PMap;
+
+public class CustomCarFlagEncoder extends CarFlagEncoder {
+    private String name;
+
+    public CustomCarFlagEncoder(PMap properties, String name) {
+        super(properties);
+        this.name = name;
+        if (!properties.getBool("block_private", true)) {
+            this.restrictedValues.remove("private");
+            this.intendedValues.add("private");
+            this.intendedValues.add("destination");
+        }
+
+        if (!properties.getBool("block_delivery", true)) {
+            this.restrictedValues.remove("delivery");
+            this.intendedValues.add("delivery");
+        }
+    }
+
+    protected final double applyMaxSpeed(ReaderWay way, double speed) {
+        double maxSpeed = this.getMaxSpeed(way);
+        return maxSpeed >= 0.0D ? Math.max(5.0D, Math.min((double)this.maxPossibleSpeed, maxSpeed)) : speed;
```

Comment:
This is....hacky. Basically, when 1.0 was merged into our fork, we started referencing a particular base jar dependency for as many classes as possible, vs what we used to do, which is store all of the source files for everything (so, we only have .java files for the things we need to add/modify, and everything else is referenced in pre-built jars). 

You'll see that in our master right now, this dependency is pointing to a [specific class](https://github.com/replicahq/graphhopper/blob/original-direction/web-bundle/src/main/java/com/graphhopper/http/GraphHopperManaged.java#L31) in that version of 1.0's jar called `CustomCarFlagEncoder` (if you click into that class in an IDE you can see it). All I did here, because I necessarily needed to add `name` to make congestion work, but didn't want to change the functionality of the code, was copy+paste that class into a new file (now in package `com.graphhopper.swl`), and added the `name` param.

Looking again, though, I agree that this code is strange (/possibly incorrect for our purposes). @michaz's commit [here](https://github.com/replicahq/graphhopper/commit/e8df8536586554935671c0311c29f9ed5ed2479b#diff-130ed9726000f7f80461f0c819f1e5c4R31) when merging in GH 1.0 moved us to using this custom flag encoder, mainly because we needed to initialize it with a `configuration` to fix a NPE. The truck flag encoder we've used for the past [8 months](https://github.com/replicahq/graphhopper/commit/d8256b1e31a08a49dec3157032cdf76f78859d0b#diff-130ed9726000f7f80461f0c819f1e5c4R25) currently also extends this class, so that's probably why he chose to use this version of `CustomCarFlagEncoder`.

Reply:
Sorry for the context dump above. The two options basically boil down to 

1. Keep things as-is, which preserves the current functionality of master (which is currently in the state of just having had GH 1.0 merged into it)

2. Remove the extra logic/functions present in this class, so that it only takes in a configuration and name (the only "custom" info we required to have the encoder function properly); in this case, it will essentially act as a default GH car flag encoder would, except for storing a name and initializing itself with a config.

I'd lean towards 2 now that I think about it, but I'd like to know Michael's opinion, given that he moved us to this specific version of the code.

Reply:
This happens when the OSM max-speed tags are interpreted. I don't fully understand this, to be honest, I copy+pasted it from our own instance.
Feel free to remove any overloaded method other than those necessary for the multiple speed profiles, and it should behave like the open-source version, which most people are fine with..

Reply:
Simplified our lives by removing everything except the constructor (giving us the necessary `properties` and `name` params), and `toString`, which is all that's needed for GH to differentiate the car flag encoders when >1 are being used.

Reply:
Technically, `TruckFlagEncoder` still naturally extends the version of `CustomCarFlagEncoder` in the GH core codebase that does this strange max-speed setting, but that's been the case since we started using the truck flag encoder I believe (?), so it's more of an accidental discovery, not a change in behavior.